lib/xcode/install.rb

```diff
@@ -58,13 +129,20 @@ def current_symlink
       File.symlink?(SYMLINK_PATH) ? SYMLINK_PATH : nil
     end
 
-    def download(version, progress, url = nil)
+    def download(version, progress, url = nil, progress_block = nil)
```

Comment:
Instead of changing the method signature, any reason not use the `&block` parameter? It's always there and can be optional.

Reply:
Great idea, will do

Reply:
I am not quite sure I understand.

My initial thought was to change the method signature to:
`def download(version, progress, url = nil, &block)`

And then calling
```
result = Curl.new.fetch(
+        url: url || xcode.url,
+        directory: CACHE_DIR,
+        cookies: url ? nil : spaceship.cookie,
+        output: dmg_file,
+        progress: progress,
+        progress_block: &block
+      )
```
But after reviewing the changes some more, it doesn't look like adding an explicit new parameter `progress_proc` will break any existing usage so it's not a concern beyond code style.